apisix/balancer.lua

```diff
@@ -55,10 +55,12 @@ local function fetch_health_nodes(upstream, checker)
     local port = upstream.checks and upstream.checks.active and upstream.checks.active.port
     local up_nodes = core.table.new(0, #nodes)
     for _, node in ipairs(nodes) do
-        local ok = checker:get_target_status(node.host, port or node.port, host)
+        local ok, err = checker:get_target_status(node.host, port or node.port, host)
         if ok then
             -- TODO filter with metadata
             up_nodes[node.host .. ":" .. node.port] = node.weight
+        elseif err then
+            core.log.error("failed to get health check target status: ", err)
```

Comment:
Just be captious, it would be better to log down the arguments `node.host, port or node.port, host` when error occurs.

Reply:
Copy ~